File: solutions/python/scrabble-score/1/scrabble_score.py
def score(word):
    sum = 0
    for letter in word.lower():
        match letter:
            case ("a"|"e"|"i"|"o"|"u"|"l"|"n"|"r"|"s"|"t"):
                sum += 1
            case ("d"|"g"):
                sum += 2
            case ("b"|"c"|"m"|"p"):
                sum += 3
            case ("f"|"h"|"v"|"w"|"y"):
                sum += 4
            case ("k"):
                sum += 5
            case ("j"|"x"):
                sum += 8
            case ("q"|"z"):
                sum += 10
    return sum

File: solutions/fsharp/scrabble-score/1/ScrabbleScore.fs
module ScrabbleScore

open System

let score word = 
    let getScore (letter: char) =
        match Char.ToLower(letter) with
        | 'a' | 'e' | 'i' | 'o' | 'u' | 'l' | 'n' | 'r' | 's' | 't' -> 1
        | 'd' | 'g' -> 2
        | 'b' | 'c' | 'm' | 'p' -> 3
        | 'f' | 'h' | 'v' | 'w' | 'y' -> 4
        | 'k' -> 5
        | 'j' | 'x' -> 8
        | 'q' | 'z' -> 10
        | _ -> 0
    word |> Seq.fold (fun s a -> s + getScore a) 0